node/silkworm/rpc/server/server_context_pool.hpp

```diff
@@ -49,23 +48,32 @@ class ServerContext {
     void stop();
 
   private:
+    //! Execute asio-grpc loop until stopped.
+    void execute_loop_agrpc();
+
     //! Execute single-threaded loop until stopped.
     template <typename WaitStrategy>
     void execute_loop_single_threaded(WaitStrategy&& wait_strategy);
 
     //! Execute multi-threaded loop until stopped.
     void execute_loop_multi_threaded();
 
-    //! The asynchronous event loop scheduler.
+    //! The unique scheduler identifier.
+    std::size_t context_id_;
+
+    //! The asio asynchronous event loop scheduler.
     std::shared_ptr<boost::asio::io_context> io_context_;
 
-    //! The work-tracking executor that keep the scheduler running.
+    //! The work-tracking executor that keep the asio scheduler running.
     boost::asio::execution::any_executor<> work_;
 
-    std::unique_ptr<grpc::ServerCompletionQueue> server_queue_;
-    std::unique_ptr<CompletionEndPoint> server_end_point_;
-    std::unique_ptr<grpc::CompletionQueue> client_queue_;
-    std::unique_ptr<CompletionEndPoint> client_end_point_;
+    //! The asio-grpc asynchronous event schedulers.
+    std::unique_ptr<agrpc::GrpcContext> server_grpc_context_;
+    std::unique_ptr<agrpc::GrpcContext> client_grpc_context_;
+
+    //! The work-tracking executors that keep the asio-grpc scheduler running.
+    boost::asio::executor_work_guard<agrpc::GrpcContext::executor_type> server_grpc_context_work_;
+    boost::asio::executor_work_guard<agrpc::GrpcContext::executor_type> client_grpc_context_work_;
```

Comment:
This code suggests that perhaps there is another abstraction hidden in the code, an abstraction with an agrpc::GrpcContext and an executor_work_guard. Maybe a ConnectionEndPoint?
In any case we can think about it in the future, let's leave it as it is, since it works.

Reply:
OK, let's see how this works for now, IMHO it's not a big deal